Todo/AppDelegate.swift

```diff
@@ -0,0 +1,420 @@
+            if object as? NSObject == conflictsLiveQuery {
+                resolveConflicts(conflictsLiveQuery!.rows)
+            }
+    }
+
+    func startConflictLiveQuery() {
+        conflictsLiveQuery = database.createAllDocumentsQuery().asLiveQuery()
+        conflictsLiveQuery!.allDocsMode = .OnlyConflicts
+        conflictsLiveQuery!.addObserver(self, forKeyPath: "rows", options: .New, context: nil)
+        conflictsLiveQuery!.start()
+    }
+
+    func stopConflictLiveQuery() {
+        conflictsLiveQuery?.removeObserver(self, forKeyPath: "rows")
+        conflictsLiveQuery?.stop()
+        conflictsLiveQuery = nil
+    }
+
+    func resolveConflicts(queryEnum: CBLQueryEnumerator?) {
+        guard let rows = queryEnum else {
+            return
+        }
+
+        for row in rows {
+            if let revs = row.conflictingRevisions! where revs.count > 1 {
+                resolveConflicts(revisions: row.conflictingRevisions)
+            }
+        }
+    }
+
+    func resolveConflicts(revisions revs: [CBLRevision]!) {
+        let defaultWinning = revs[0]
+        let type = defaultWinning["type"] as? String
+        if type == "task-list" || type == "task-list-user" {
+            let props = defaultWinning.userProperties
+            let image = defaultWinning.attachmentNamed("image")
+            resolveConflicts(revisions: revs, withProps: props, andImage: image)
+        } else if type == "task" {
+            let merged = nWayMergeConflicts(revs)
+            resolveConflicts(revisions: revs, withProps: merged.props, andImage: merged.image)
+        }
+    }
+
+    func resolveConflicts(revisions revs: [CBLRevision]!, withProps props: [String: AnyObject]?,
+        andImage image: CBLAttachment?) {
+            database.inTransaction { () -> Bool in
+                var i = 0
+                for rev in revs as! [CBLSavedRevision] {
+                    let newRev = rev.createRevision()
+                    if (i == 0) { // Default winning revision
+                        newRev.userProperties = props
+                        if rev.attachmentNamed("image") != image {
+                            newRev.setAttachmentNamed("image", withContentType: "image/jpg",
+                                content: image?.content)
+                        }
+                    } else {
+                        newRev.isDeletion = true
+                    }
+
+                    do {
+                        try newRev.saveAllowingConflict()
+                    } catch let error as NSError {
+                        NSLog("Cannot resolve conflicts with error: %@", error)
+                        return false
+                    }
+                    i++
+                }
+                return true
+            }
+    }
+
+    func nWayMergeConflicts(revs: [CBLRevision]!) ->
+        (props: [String: AnyObject]!, image: CBLAttachment?) {
+            let parent = findCommonParent(revs)
```

Comment:
I think you can use `guard let parent = ...` here, and avoid having to make `parent` an optional. Then you don't have to use `!` after it down below.